server/planning/assignments.py

```diff
@@ -0,0 +1,191 @@
+            get_resource_service('planning').sync_coverages([{'planning_item': original.get('planning_item')}])
+
+    def system_update(self, id, updates, original):
+        super().system_update(id, updates, original)
+        self.notify('assignments:updated', updates, original)
+
+    def is_assignment_modified(self, updates, original):
+        """Checks whether the assignment is modified or not"""
+        updates_assigned_to = updates.get('assigned_to') or {}
+        original_assigned_to = original.get('assigned_to') or {}
+        return updates_assigned_to.get('desk') != original_assigned_to.get('desk') or \
+               updates_assigned_to.get('user') != original_assigned_to.get('user')
+
+    def send_assignment_notification(self, updates, original=None):
+        """Set the assignment information and send notification
+
+        :param dict doc: Updates related to assignments
+        """
+        if not original:
+            original = {}
+
+        if not self.is_assignment_modified(updates, original):
+            return
+
+        assigned_to = updates.get('assigned_to')
+        user = get_user()
+        if assigned_to.get('user'):
+            # Done to avoid fetching users data for every assignment
+            # Because user assigned can also be a provider whose qcode
+            # might be an invalid GUID, check if the user assigned is a valid user (GUID)
+            # However, in a rare case where qcode of a provider is a valid GUID,
+            # This will create activity records - inappropirate
+            if ObjectId.is_valid(assigned_to.get('user')):
+                add_activity(ACTIVITY_UPDATE,
+                             '{{assignor}} assigned a coverage to {{assignee}}',
+                             self.datasource,
+                             notify=[assigned_to.get('user')],
+                             assignor=user.get('username')
+                             if str(user.get(config.ID_FIELD, None)) != assigned_to.get('user') else 'You',
+                             assignee='you'
+                             if str(user.get(config.ID_FIELD, None)) != assigned_to.get('user') else 'yourself')
+
+assignments_schema = {
+    # Audit Information
+    'original_creator': metadata_schema['original_creator'],
+    'version_creator': metadata_schema['version_creator'],
+    'firstcreated': metadata_schema['firstcreated'],
+    'versioncreated': metadata_schema['versioncreated'],
+
+    # Assignment details
+    'coverage_item': coverage_type,
+    'planning_item': planning_type,
+
+    'assigned_to': {
+        'type': 'dict',
+        'schema': {
+            'desk': {'type': 'string', 'nullable': True, 'mapping': not_analyzed},
+            'user': {'type': 'string', 'nullable': True, 'mapping': not_analyzed},
+            'assigned_by': {'type': 'string', 'mapping': not_analyzed},
+            'assigned_date': {'type': 'datetime'},
+            'state': {'type': 'string', 'mapping': not_analyzed},
```

Comment:
It would be good to set `allowed` to `assignment_state` for the state schema to enforce allowed values here